pyccel/codegen/printing/fcode.py

```diff
@@ -659,8 +659,12 @@ def _print_NumpyNorm(self, expr):
         """Fortran print."""
         arg = PythonAbs(expr.arg) if isinstance(expr.arg.dtype, NativeComplex) else expr.arg
         if expr.axis:
-            axis = str(abs(expr.axis.python_value - arg.rank)) + '_C_DOUBLE'
-            code = 'Norm2({},{})'.format(self._print(arg), axis)
+            axis = expr.axis
+            if arg.order != 'F':
+                axis = PythonAbs(PyccelMinus(expr.axis, LiteralInteger(arg.rank)))
+            else:
+                axis = LiteralInteger(expr.axis.python_value + 1)
```

Comment:
Why do you need a `+1` here, but not above?

Reply:
i'm trying to understand why does not give the same result as Fortran at Fortran ordered array with many ways and this is one of them, after some testing i don't think this can help anymore.

Reply:
given that Python gives the same result with both `C` and `F` order.

Reply:
@EmilyBourne 
> Why do you need a `+1` here, but not above?

`arg.rank` is the size not the ubound so the `+1` is already contained in this variable